### PR TITLE
[OPIK-2355] [BE] Add annotation queue item management endpoints

### PR DESCRIPTION
**Pull request by BorisTkachenko**
## Details

This PR implements endpoints for managing annotation queue items, enabling users to add and remove traces/threads from annotation queues for human review workflows.

**Key Features Implemented:**
- **New AnnotationQueueItemIds DTO** for handling sets of UUIDs in API requests
- **POST /v1/private/annotation-queues/{id}/items/add endpoint** for adding items to annotation queues
- **POST /v1/private/annotation-queues/{id}/items/delete endpoint** for removing items from annotation queues
- **ClickHouse table migration** for annotation queue items storage
- **Comprehensive test coverage** including unit and integration tests

**Architecture Implementation:**
- **Resource Layer**: New endpoints in `AnnotationQueuesResource` with proper validation and OpenAPI documentation
- **Service Layer**: Enhanced `AnnotationQueueService` with reactive item management methods
- **Data Layer**: New ClickHouse table `annotation_queue_items` with proper clustering and ordering

**Database Changes:**
- **ClickHouse Migration**: Added `annotation_queue_items` table for tracking queue membership
- **Schema Design**: Workspace/project/queue/item organization with timestamps and audit fields
- **Performance**: Proper clustering and ordering for efficient queries

## Change checklist
- [x] User facing
- [ ] Documentation update

## Issues
- OPIK-2355

## Testing

**Integration Tests:**
- `AnnotationQueuesResourceTest`: Added `AnnotationQueueItemManagement` nested test class with comprehensive coverage
- Tests include: successful addition/removal, validation, authentication, and error scenarios

**Unit Tests:**
- Updated existing test infrastructure to support new endpoints
- Enhanced `AnnotationQueuesResourceClient` for integration testing

**Test Coverage:**
- Success cases: adding and removing annotation queue items
- Error cases: invalid inputs, authentication failures
- Edge cases: empty requests, non-existent queues

## Documentation

**OpenAPI Documentation:**
- Complete endpoint documentation for both add and remove operations
- `AnnotationQueueItemIds` schema with validation constraints (1-1000 items)
- Proper HTTP status codes and error response documentation

**Database Documentation:**
- ClickHouse migration script with rollback support
- Table schema documentation with field explanations

This implementation enables efficient management of annotation queue items, supporting the human annotation workflow by allowing dynamic addition and removal of traces and threads from queues.

### CODE REVIEW

File: apps/opik-backend/src/main/java/com/comet/opik/api/AnnotationQueueItemIds.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+import java.util.Set;
+import java.util.UUID;
+
+@Builder(toBuilder = true)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
+public record AnnotationQueueItemIds(
+        @NotNull @Size(min = 1, max = 1000) Set<UUID> ids) {
+}
```
Comment (idoberko2) [inline apps/opik-backend/src/main/java/com/comet/opik/api/AnnotationQueueItemIds.java]: Very minor: blank line at the end of the file

File: apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/AnnotationQueuesResourceTest.java [hunk 2]
```diff
@@ -143,4 +142,133 @@ void createAnnotationQueueWhenProjectIdNullShouldReject() {
...
+                    new LinkedHashSet<>(List.of(annotationQueue)), API_KEY, TEST_WORKSPACE, HttpStatus.SC_NO_CONTENT);
+
+            // Generate some item IDs to add
+            var itemIds = Set.of(
+                    UUID.randomUUID(),
+                    UUID.randomUUID(),
+                    UUID.randomUUID());
+
+            // When & Then
+            annotationQueuesResourceClient.addItemsToAnnotationQueue(
```
Comment (idoberko2) [inline apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/AnnotationQueuesResourceTest.java:178]: Seems like there's a missing assertion to verify the items were added correctly

File: apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/AnnotationQueuesResourceTest.java [hunk 3]
```diff
@@ -143,4 +142,133 @@ void createAnnotationQueueWhenProjectIdNullShouldReject() {
...
+                    UUID.randomUUID(),
+                    UUID.randomUUID());
+
+            // Add items first
+            annotationQueuesResourceClient.addItemsToAnnotationQueue(
+                    annotationQueue.id(), itemIds, API_KEY, TEST_WORKSPACE, HttpStatus.SC_NO_CONTENT);
+
+            // When & Then - Remove the items
+            annotationQueuesResourceClient.removeItemsFromAnnotationQueue(
+                    annotationQueue.id(), itemIds, API_KEY, TEST_WORKSPACE, HttpStatus.SC_NO_CONTENT);
```
Comment (idoberko2) [inline apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/AnnotationQueuesResourceTest.java:212]: Again missing an assertion here to verify that the items were indeed removed
